lib/wiki_api.rb

```diff
@@ -97,46 +101,36 @@ def mediawiki(action, query)
     tries ||= 3
     @mediawiki = api_client
     @mediawiki.send(action, query)
-  rescue MediawikiApi::ApiError => e
-    log_error e, action, query
   rescue StandardError => e
     tries -= 1
-    log_error e, action, query
-    handle_non_api_error(e)
-    retry if tries >= 0
-    Raven.capture_exception e, level: 'warning'
-    return nil # Do not return a Raven object
+    # Raise unknown errors.
+    # Continue for typical errors so that the request can be retried, but wait
+    # a short bit in the case of 429 — too many request — errors.
+    sleep 1 if too_many_requests?(e)
+    retry unless tries.zero?
+    invoke_error_handling_tasks(e, action, query)
+    raise e unless TYPICAL_ERRORS.include?(e.class)
```

Comment:
I think the raising of unexpected errors could also be done in the error-handling module; this would probably mean passing in the relevant set of `TYPICAL_ERRORS` as an argument in the error handling code (or getting it dynamically from the class that the error handling method is being invoked from).

Reply:
Done!